Docs/troubleshooting.md

```diff
@@ -0,0 +1,236 @@
+| LOG_LEVEL | "1" or "Debug"       | Everything is logged, including the up- and downstream messages to the packet forwarder. |
+|           | "2" or "Information" | Errors and information are logged.                                                       |
+|           | "3" or "Error"       | Only errors are logged. (default if omitted)                                             |
+
+For production environments, the **LOG_LEVEL** should be set to **Error**.
+
+Setting **LOG_LEVEL** to **Debug** causes a lot of messages to be generated. Make sure to set **LOG_TO_HUB** to **false** in this case.
+
+| Variable   | Value | Explanation                                          |
+|------------|-------|------------------------------------------------------|
+| LOG_TO_HUB | true  | Log info are sent from the module to IoT Hub.        |
+|            | false | Log info is not sent to IoT Hub (default if omitted) |
+
+You can use VSCode, [IoTHub explorer](https://github.com/Azure/iothub-explorer) or [Device Explorer](https://github.com/Azure/azure-iot-sdk-csharp/tree/master/tools/DeviceExplorer) to monitor the log messages directly in IoT Hub if **LOG_TO_HUB** is set to **true**.
+
+Log in to the gateway and use `sudo docker logs LoRaWanNetworkSrvModule -f` to follow the logs if you are not logging to IoT Hub.
+
+| Variable       | Value | Explanation                             |
+|----------------|-------|-----------------------------------------|
+| LOG_TO_CONSOLE | true  | Log to docker logs (default if omitted) |
+|                | false | Does not log to docker logs             |
+
+## Customize the solution
+
+Have a look at the [LoRaEngine folder](/LoRaEngine) for more in details explanation.
+
+## Cloud to device message
+
+The solution support sending Cloud to device (C2D) messages to LoRa messages using [standard IoT Hub Sdks](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-c2d). Cloud to device messages require a Fport message property being set or it will be refused (as shown in the figure below from the Azure Portal). 
+
+![C2D portal](/Docs/Pictures/cloudtodevice.png)
+
+The following tools can be used to send cloud to devices messages from Azure :
+
+* [Azure Portal](http://portal.azure.com) -> IoT Hub -> Devices -> message to device
+* [Device Explorer](https://github.com/Azure/azure-iot-sdk-csharp/tree/master/tools/DeviceExplorer)
+* [Visual Studio Code IoT Hub Extension](https://marketplace.visualstudio.com/items?itemName=vsciot-vscode.azure-iot-toolkit) 
+
+It is possible to add a 'Confirmed' message property set to true,in order to send the C2D message as ConfirmedDataDown to the LoRa device (as in picture above and below). You can enable additional message tracking options by setting the C2D message id to a value (C2D message ID is automatically populated with the Device Explorer tool used in the image below). 
```

Comment:
Change the first sentence in a way, that "in order" part comes first --> 

In order to send the C2D message as ConfirmedDataDown to the LoRa device (as in picture above and below), it is possible to add a 'Confirmed' message property set to true. 

The second sentence seems to be missing something.... by setting the C2D message id to a value... --> what value?